guides/oidc/oidc.md

```diff
@@ -0,0 +1,116 @@
+# OpenID Connect
+
+The OpenID Connect (OIDC) feature allows third-party applications to verify the identity of the end-user and to obtain basic user profile information. Instead of managing usernames and passwords directly **trento-web console will rely on an identity provider (IDP) to authenticate users**.
+
+## Enabling OIDC
+
+The OIDC feature is **disabled by default**.
+
+Provide the following environment variable to enable OIDC feature when starting Trento.
+
+```
+ENABLE_OIDC=true
+OIDC_CLIENT_ID=<<OIDC_CLIENT_ID>>
+OIDC_CLIENT_SECRET=<<OIDC_CLIENT_SECRET>>
+OIDC_BASE_URL=<<OIDC_BASE_URL>>
+OIDC_CALLBACK_URL=<<OIDC_CALLBACK_URL>>
+```
+
+## Enabling OIDC in Development
+
+Enable OpenID Connect (OIDC) in the development environment using Docker and [Keycloak](https://github.com/keycloak/keycloak) as a simulated Identity Provider (IDP).
+
+### Enable OIDC in dev environment by editing `config/dev.exs`
+
+Before:
+
+```
+config :trento, :oidc,
+  enabled: false,
+  callback_url: "http://localhost:4000/auth/oidc_callback"
+```
+
+After:
+
+```
+config :trento, :oidc,
+  enabled: true,
+  callback_url: "http://localhost:4000/auth/oidc_callback"
+```
+
+### Starting the Keycloak Identity Provider
+
+We use a custom Docker profile to start Keycloak as the Identity Provider (IDP) for local development.
+
+Start the Docker containers with the `IDP` profile:
+
+```
+docker-compose --profile idp up
+```
+
+### Starting Trento Web console
```

Comment:
Do we need to reference this?
Just mention to start web as usual, the same for accessing.
If you want to explain the usage of the keycloak console, i would use other 3rd chapter, because it is not really related on how to operate trento

Reply:
Restructured this part, what do you think?